### ISSUE TITLE
Lumberjack sapling use needs work

### ISSUE THREAD

**Issue by MattBeard** [ldtteam/minecolonies#1634]
### Minecolonies version

version 1.12.2-0.8.4802 or curse 7.5.6522.42411

### Expected behavior

The lumberjack should be able to find saplings when required, even if they need to fetch them from their chest or request from deliveryman.

Ideal behaviour:
1) If no sapling is available at the end of chopping: After a waiting a while while (or slowly while waiting) scan the area within about 10 blocks of the base of the tree for saplings and move to collect if they are found.

2) If this does not work after a certain amount of time, check the lumberjack chest for the right sapling.

3) If not found in the chest, request a delivery.

4) Only "give up" as a last resort.

Additionally: There needs to be some way to scan the area a little above/below the chopping site for saplings and somehow collect them.

### Actual behaviour

Often the lumber jack will end up "giving up" on planting a sapling for a tree because they don't have one. The newer improved behaviour of walking around while waiting helps a little, but certain trees such as the dark oak (which requires 4 saplings and does not drop all that many) often end up causing a problem. Jungle trees seem to be even worse as I have noticed that saplings often drop and sit in the canopy or on top of the trunk for so long they de-spawn.

This can be heartbreaking when you have spent hours on a long trip to find saplings for a rare tree and bring them back and plant them, only to notice later that they are all gone and not replanted.

### Steps to reproduce the problem

1. Plant a few dark oak and/or jungle trees
2. Let the lumberjack chop them down
3. Watch them eventually run out of saplings
4. For extra fun: Set up a warehouse and deliveryman and watch the warehouse fill up with the saplings that your lumberjack is missing! Especially true if your guards patrol the area and pick them up.

**Comment by Asherslab** [comment]
The warehouse / DMan are both getting a serious overhaul (Called the RS) this *should* fix what you mention with 4. (steps to reproduce). otherwise, yes they could use a bit of a change

**Comment by PhoenixRiver** [comment]
This is also a problem with Natura install and Biomes O' Plenty due to there being a bunch of trees with low drop rates and possibly some of the trees having reduced drop rates - the Wooden Warehouse in Mine Colonies becomes very hard to build if you have to import the spruce wood but the drops from the tree often land on lower leaves and the drop rate is pretty bad.

**Comment by PhoenixRiver** [comment]
Now running minecolonies-universal-1.12.2-0.8.4932 and lumberjack still having problems to the point that he isn't even collecting enough to keep oaks planted on a plantation I have created around his hut as below [/// : hut, - : nothing, . : tree]:

------------------///////////------------------
------------------///////////------------------
--.-----.-----.---///////////---.-----.-----.--
------------------///////////------------------
------------------///////////------------------
------------------///////////------------------
------------------///////////------------------
------------------///////////------------------
--.-----.-----.---///////////---.-----.-----.--
------------------///////////------------------
------------------///////////------------------
------------------///////////------------------
------------------///////////------------------
------------------///////////------------------
--.-----.-----.---///////////---.-----.-----.--
------------------///////////------------------
------------------///////////------------------
------------------///////////------------------
--------------------------------------------
--------------------------------------------
--.-----.-----.-----.-----.-----.-----.-----.--
--------------------------------------------
--------------------------------------------
--------------------------------------------
--------------------------------------------
--------------------------------------------
--.-----.-----.-----.-----.-----.-----.-----.--
--------------------------------------------
--------------------------------------------
--------------------------------------------
--------------------------------------------
--------------------------------------------
--.-----.-----.-----.-----.-----.-----.-----.--
--------------------------------------------
--------------------------------------------

Essentially a grid of trees with a gap of five between each tree - this means the lumberjack covers a lot of the ground where saplings fall regularly, despite this he still is running out of oak ones to replant and so there are now missing trees that I have to replace.

**Comment by Raycoms** [comment]
I will have to go now, but I think the problem is that if he has a sapling
he doesn't try to gather and that's why he ends up having none after a
while as well. I'll add that he tries to gather the saplings after
replanting always.
I also will add that he checks in his hut for the right sapling after a
shorter delay.

2017-11-21 15:53 GMT+00:00 PhoenixRiver <notifications@github.com>:

**Comment by PhoenixRiver** [comment]
Those would make a big difference I think :) - thank you

**Comment by Neferdias** [comment]
I have noticed that my lumberjack won't replant birch or spruce saplings, even if they are in the chest. He'll just give up trying to plant them after standing around for 5 minutes. If I go grab one from the chest and throw it at his feet he'll plant it and carry on his way.

**Comment by Raycoms** [comment]
Check out this Lumberjack:
https://s3.eu-central-1.amazonaws.com/minecolonies/Minecolonies/Minecolonies_Branches/5088/minecolonies-universal-1.12.2-0.8.5088.jar

Anything to complain about him?

2017-11-21 22:58 GMT+00:00 Neferdias <notifications@github.com>:

**Comment by Neferdias** [comment]
As far as I can tell it's now working as intended. (I'm only working off of what versions are posted to twitch/curse gaming app that i can get working, so that might be a me problem mostly)

**Comment by PhoenixRiver** [comment]
Okay, the lumberjack worked well on the first and second day and it
definitely solved the problem as he collected a lot of saplings... too
many is not a bad thing I can just burn them for fuel! :)  buuuut...
the farmers just stopped moving around; when recalled they just stood
by their chest not apparently farming at all - they didn't seem to
collect any crops at all over three days (there are four farmers at
level two with four working fields and two fields of barley). The
builder still built and the deliveryman still delivered. However on
the third day the farmers, builder and lumberjack didn't leave their
houses, only the unemployed person wandered and the deliveryman did
his job...

TLDR; Lumberjack works better until the third day, farmers broken and
possibly builder too also on third day unless she just doesn't wander
unless she has a job. Unemployed people wander around just fine! :)

**Comment by Raycoms** [comment]
Can you send me the log?
Did you turn off the daylight cycles?

2017-11-22 15:56 GMT+00:00 PhoenixRiver <notifications@github.com>:

**Comment by PhoenixRiver** [comment]
The only mine colonies relevant entries are the missing sound stuff
and requests for more fields.

I didn't turn off day night cycles I just sleep at night :) - I have
no town wall yet! :)

**Comment by MattBeard** [comment]
Has the lumberjack been updated to plant 4 jungle saplings when they only have 1 in their inventory? Seem a, um, _creative_, fix!

**Comment by Raycoms** [comment]
No I haven't change anything in that code

**Comment by MattBeard** [comment]
Odd, because my lumberjack had no saplings, I gave him 1 jungle - and he them planted 4

**Comment by Raycoms** [comment]
He didn't get them elsewhere?

**Comment by MattBeard** [comment]
Pretty sure not. I will keep an eye on this!
